### DISCUSSION TITLE
Detect device capabilities?

### DISCUSSION THREAD

**Discussion by lanwin** [omerfaruk-aran/esphome_samsung_hvac_bus#67] [General]
@Foxhill67 did you found a way to detect the device capabilities? Some AC have features like Windfree or Quite mode some dont.

**Comment by Foxhill67** [comment]
Not really. With my heat pump I just activated logging of all messages via MQTT, and compared the NASA message codes with the list on https://wiki.myehs.eu/wiki/NASA_Protocol#. I assume each device will only publish relevant messages?
